### PR TITLE
Shallow Load Project Types when loading all

### LINKED ISSUE #610: Calling help in a project type loads all project type commands
# Issue summary

Calling help in a project type context actually loads the commands of all project types. This is a problem in the case that one project type does not have a command that another project type does. 

## Expected behavior

Calling `help` in a project type context should only return commands from that project type.

## Actual behavior

In a script project, we get the help for commands like `generate`, `populate`, and `tunnel` from rails/node projects.

In a node project, we get the help for commands specific to scripts. So `disable`, `enable`, and `test` commands. 

```
Use shopify help <command> to display detailed information about a specific command.

Available core commands:

connect: Connect a Shopify App CLI project. Restores the ENV file.
  Usage: shopify connect

create: Create a new project.
  Usage: shopify create [ node | rails ]

logout: Log out of a currently authenticated Organization and Shop, or clear invalid credentials
  Usage: shopify logout

update: Update Shopify App CLI.
  Usage: shopify update

Project: example-node (Node.js App)
Available commands for Node.js App projects:

deploy: Deploy the current Node project to a hosting service. Heroku (https://www.heroku.com) is currently the only option, but more will be added in the future.
  Usage: shopify deploy [ heroku ]

disable: Turn off script in development store.
  Usage: shopify disable --API_key=<API_key> --shop_domain=<my_store.myshopify.com>

enable: Turn on script in development store.
  Usage: shopify enable --API_key=<API_key> --shop_domain=<my_store.myshopify.com>

generate: Generate code in your Node project. Supports generating new billing API calls, new pages, or new webhooks.
  Usage: shopify generate [ billing | page | webhook ]

open: Open your local development app in the default browser.
  Usage: shopify open

populate: Populate your Shopify development store with example customers, orders, or products.
  Usage: shopify populate [ customers | draftorders | products ]

serve: Start a local development node server for your project, as well as a public ngrok tunnel to your localhost.
  Usage: shopify serve

test: Runs unit tests on your script.
  Usage: shopify test

tunnel: Start or stop an http tunnel to your local development app using ngrok.
  Usage: shopify tunnel [ auth | start | stop ]
```

### PR DESCRIPTION
**Pull request by tanema**
### WHY are these changes introduced?

Fixes #610

This fixes help commands outputting in different project types.

### WHAT is this pull request doing?
I have changed ProjectType to only load the creators and messages when loading all of the project types because that is the only information needed during processes that need all of the information.

### DEAR GOD WHAT HAVE YOU DONE
It seems like once I stopped loading all of the commands for all of the projects then the tests got more flakey. I believe that this is because some commands were loading over each other and using `run_cmd` would call the other implemented commands! *oh shit* 
 
So what I have done here is for quite a few tests
- Added a pattern introduced by @jacobsteves where each project type has it's own test helper that it imports, loading their own project type
- Refactored a lot of the calling code to call the commands directly instead of using run_cmd (some progress on making real unit tests)

This has made the tests a lot more stable for me once again!

### CODE REVIEW

**Comment by tanema** [comment]
I think loading shallow has actually made the tests more flaky which means some of our tests were not loading all of the classes that they need! I might actually introduce a pattern into our project_types that the scripts started to try and make this more reliable.

File: lib/shopify-cli/project_type.rb [hunk 1]
```diff
@@ -54,13 +58,15 @@ def hidden_project_type
...
         Context.new.abort(
           Context.message('core.project_type.error.cannot_override_core', cmd, const)
         ) if Commands.core_command?(cmd)
         Commands.register(const, cmd)
       end
 
       def register_task(task, name)
+        return if project_load_shallow
         Task::Registry.add(const_get(task), name)
       end
```
Comment (paulomarg) [inline lib/shopify-cli/project_type.rb:72]: Should we also not load the messages if we're shallow loading? That would reduce the overhead of reading the files.

File: test/shopify-cli/project_type_test.rb [hunk 2]
```diff
@@ -38,5 +38,11 @@ def test_duplicate_command
         ProjectType.register_command('Nonsense::Module::Help', 'help')
       end
     end
+
+    def test_register_command_does_not_call_if_shallow
```
Comment (paulomarg) [inline test/shopify-cli/project_type_test.rb:42]: Do we want a similar test for tasks?

File: test/project_types/node/commands/populate/customer_test.rb [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
   module Commands
     module PopulateTests
       class CustomerTest < MiniTest::Test
         include TestHelpers::Schema
 
-        def setup
-          super
-          ShopifyCli::ProjectType.load_type(:node)
-        end
-
```
Comment (ayronshopify) [inline test/project_types/node/commands/populate/customer_test.rb:13]: I noticed that the test in this file is still running `run_cmd` are you only updating some of the tests now?

File: test/project_types/node/test_helper.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+require "test_helper"
+
+ShopifyCli::ProjectType.load_type(:node)
```
Comment (ayronshopify) [inline test/project_types/node/test_helper.rb:4]: 😍

**Pull request merged**